core/src/utils/theme.ts

```diff
@@ -6,15 +6,20 @@ export const hostContext = (selector: string, el: HTMLElement): boolean => {
 };
 
 /**
- * Create the mode and color classes for the component based on the classes passed in
+ * Create the mode and color classes for the component based on the classes passed in and the theme being used
  */
 export const createColorClasses = (color: Color | undefined | null, cssClassMap: CssClassMap): CssClassMap => {
   return typeof color === 'string' && color.length > 0
-    ? {
-        'ion-color': true,
-        [`ion-color-${color}`]: true,
-        ...cssClassMap,
-      }
+    ? cssClassMap.ionic === true
```

Comment:
This is a pretty big divergence from the way we have generated & used colors. Was there a ticket or discussion surrounding this?

Reply:
There was not, but we can certainly bring this to the rest of the team!

But with Ionic Theme we cannot use this function as it was, as it will return colors from the current md/ios Design System, not the new one. 

Besides, on the new Design System we have a much wider array of color options and the expectaction would be the developers could use the new colors we are offering with the tokens and new Design System on the color attribute on each component (only when using the new Ionic Theme).

Is there any other way you would see this being done? Or is there any impact that I'm missing for this?

Reply:
I highly recommend that a spike is created first to determine why we need this, who is requesting this change, and any potential risks, etc. After the spike is completed, we can return to this PR if need be.

Reply:
This makes sense to me, I created [this spike](https://outsystemsrd.atlassian.net/browse/ROU-10778) to tackle it. 

Thanks for the feedback 👍